badger_db.go

```diff
@@ -279,19 +279,22 @@ func (i *badgerDBIterator) Valid() bool {
 	return true
 }
 
+// Key implements Iterator.
```

Comment:
```suggestion
```

I'd remove these comments.

Reply:
🤔 I would leave them, because I think they give readers some additional context about the functions. 
Granted, in the context of this repo it's clear what the functions refer to. 
Still, I would argue it doesn't hurt to leave extra-info in.